src/hotspot/share/gc/parallel/mutableSpace.cpp

```diff
@@ -60,7 +60,7 @@ void MutableSpace::numa_setup_pages(MemRegion mr, size_t page_size, bool clear_s
       size_t size = pointer_delta(end, start, sizeof(char));
       if (clear_space) {
         // Prefer page reallocation to migration.
-        os::free_memory((char*)start, size, page_size);
+        os::free_memory((char*)start, size, page_size, mtGC);
```

Comment:
Not at all obvious where the corresponding allocation sets the type as mtGC. ??

Reply:
We don't, and I am not sure this is right. AFAICS this API is used on java heap, in ParallelGC. So, that should be mtJavaHeap, I think.

Note that I would like and plan to simplify this API, if possible remove both the page size and the NMT flags. See https://bugs.openjdk.org/browse/JDK-8330144. (the tricky part is to make sure the proposed Linux alternative works with large pages and on old enough kernels)

Reply:
`os::free_memory` on Linux, re-commits the region to discard the existing committed memory. So MEMFLAGS is needed here to pass down to the `os::commit_memory` there.
Since it is actually an uncommit, can we use `mtNone` instead in the `pd_free_memory`? Or define a new `mtDontCare`, for example?